pkg/server/domain/service/webhook.go

```diff
@@ -187,28 +189,96 @@ func (c *webhookServiceImpl) patchComponentProperties(ctx context.Context, compo
 }
 
 func (c *customHandlerImpl) handle(ctx context.Context, webhookTrigger *model.ApplicationTrigger, app *model.Application) (interface{}, error) {
-	for comp, properties := range c.req.Upgrade {
-		component := &model.ApplicationComponent{
+	workflow := &model.Workflow{
+		AppPrimaryKey: webhookTrigger.AppPrimaryKey,
+		Name:          webhookTrigger.WorkflowName,
+	}
+	if err := c.w.Store.Get(ctx, workflow); err != nil {
+		return nil, err
+	}
+	switch c.req.Action {
```

Comment:
So this is a break change I think, if action is not `execute`, errors will be returned. But that break the origin behavior. Please make `execute` choice fallthough the default. If no action means error I guess the origin e2e test will fail, please check

Reply:
```
	case "execute", "":
		{
			for comp, properties := range c.req.Upgrade {
				component := &model.ApplicationComponent{
```

Here the code is performing execute operation when action is either `execute` or `null` . So, I don't think there will be any problem when there is no action.

Reply:
I see! I miss the empty string. Wondering why the test is not break...LOL.